horace_core/sqw/@sqw/tobyfit_DGfermi_resconv.m

```diff
@@ -160,136 +239,79 @@
     moderator_table.object_store = moderator;
 end
 
-reset_state=caller.reset_state;
-for i=1:numel(ind)
-    % Get index of workspace into lookup tables
-    iw=ind(i);
 
-    % Set random number generator if necessary, and save if required for later
-    if reset_state
-        if ~isempty(state_in{i})
-            rng(state_in{i})
-        end
-    else
-        state_out{i} = rng;     % capture the random number generator state
-    end
-
-    % Create pointers to parts of lookup structure for the current dataset
-    x0=lookup.x0{iw};
-    xa=lookup.xa{iw};
-    x1=lookup.x1{iw};
-    thetam=lookup.thetam{iw};
-    angvel=lookup.angvel{iw};
-    ki=lookup.ki{iw};
-    kf=lookup.kf{iw};
-    s_mat=lookup.s_mat{iw};
-    spec_to_rlu=lookup.spec_to_rlu{iw};
-    alatt=lookup.alatt{iw};
-    angdeg=lookup.angdeg{iw};
-    is_mosaic=lookup.is_mosaic{iw};
-    dt=lookup.dt{iw};
-    qw=lookup.qw{iw};
-
-    % Run and detector for each pixel
-    npix = win(i).pix.num_pixels;
-    [irun, idet] = parse_pixel_indices(win(i));     % returns column vectors
-
-    % Get detector information for each pixel in the sqw object
-    % size(x2) = [npix,1], size(d_mat) = [3,3,npix], size(f_mat) = [3,3,npix]
-    % and size(detdcn) = [3,npix]
-    [x2, detdcn, d_mat, f_mat] = detector_table.func_eval_ind (iw, irun, idet, @detector_info);
-
-    % Catch case of refining crystal orientation
-    if refine_crystal
-        % Strip out crystal refinement parameters and reorient datasets
-        [win(i), pars{1}] = refine_crystal_strip_pars (win(i), xtal, pars{1});
-
-        % Update s_mat, spec_to_rlu, alatt and angdeg because in general the
-        % crystal orientation and lattice parameters will have changed
-        [~,s_mat,spec_to_rlu,alatt,angdeg]=sample_coords_to_spec_to_rlu(win(i).experiment_info);
-
-        % Recompute Q because crystal orientation will have changed (don't need to update qw{4})
-        qw(1:3) = calculate_q (ki(irun), kf, detdcn, spec_to_rlu(:,:,irun));
+% Recompute Q on-the-fly
```

Comment:
Performing in-the-fly calculation of q has a pleasing side effect. Because calculate_q is no longer called by tobyfit_DGfermi_resconv_init, the file ...\Horace\horace_core\Tobyfit\calculate_q.m can be deleted. The present method of sqw (i.e. tobyfit_DGfermi_resconv) uses the identical file in the @sqw private folder ...\Horace\horace_core\sqw\@sqw\private\calculate_q.m. We can therefore keep the sqw private function,  delete the copy in  ...\Horace\horace_core\Tobyfit, and thereby eliminate a technical debt banana skin.